docs/developers/repository-classes.md

```diff
@@ -0,0 +1,172 @@
+inside the `targets` folder are tracked and secured by TUF.
+
+Instances of the `AuthenticationRepository` are created by passing the same arguments as to `GitRepository` (`library_dir`, `name`, `urls`, `custom`, `default_branch`, `allow_unsafe` and `path` which can replace `library_dir` and `name` combination), as well as some optional additional arguments:
+- `conf_directory_root` - path to the directory where the `last_validated_commit`will be stored. `last_validate_commit` is generated by the updater after a successful update.
+- `out_of_band_authentication` - manually specified initial commit, used during the update process to validate the first commit
+- `hosts` -  host data, specified using the `hosts.json` file. Hosts of the current repo can be specified in its
+parent's repo (meaning that this repo is listed in the parent's `dependencies.json`), or it can be specified in hosts.
+json contained by the repo itself. If hosts data is defined in the parent, it can be propagated to the contained
+repos.
+
+While in TAF's `Repository` class target files have no special meaning (it is only important that their actual states
+match the information listed in the corresponding metadata file), `AuthenticationRepository`'s target file consist
+of files which have a special meaning to the framework (like `repositories.json`) and target files which are used
+to validate target repositories. So, this class also contains methods which are specific to the authentication
+repositories as defined in TAF, and not just to TUF or communication with Git. One such widely used method is
+`sorted_commits_and_branches_per_repositories` return a dictionary consisting of branches and commits belonging to it
+for every target repository. The other is `targets_at_revisions`, which returns contents of all target files at
+revision corresponding to the specified commit.
+
+## `repositoriesdb`
+
+The purpose of this module is to automatically instantiate all target or linked authentication repositories given an authentication
+repository and, optionally, a list of commits. When a repository is instantiated, it is added to a dictionary ("database") and can be accessed when needed.
+The automatic creation of repositories relies on reading `repositories.json` and `targets.json` when instantiating
+target repositories and `dependencies.json` when creating linked authentication repositories. If commits are not
+specified, only the head of the specified authentication repository will be used. This is the most common use case as
+it is rarely needed to instantiate repositories based on some older state of the relevant files.
+
+By default, create objects will be of the `GitRepository` (targets) and `AuthenticationRepository` (linked
+authentication repositories) types. It is possible to specify a different class, as lons as it is derived from
+`GitRepository`/`AuthenticationRepository`. If not all repositories should be of the same type, it is possible to
```

Comment:
suggestion: is this more clear? `type` seems a bit ambiguous
```suggestion
`GitRepository`/`AuthenticationRepository`. If some target repository instance has to be a different derived class, it is possible to
```

Reply:
Love this, like extracting NOT in front of the sentence